### PR TITLE
Add send upstream functionality for IOS

### PR DESCRIPTION
**Pull request by mpgxvii**
- Adds upstream functionality for IOS `cordova-plugin-fcm`
- Adds setSenderId() for sending message upstream

### CODE REVIEW

File: src/ios/FCMPlugin.m [hunk 1]
```diff
@@ -106,14 +111,41 @@ -(void) notifyOfTokenRefresh:(NSString *)token
...
     if ([self.webView respondsToSelector:@selector(stringByEvaluatingJavaScriptFromString:)]) {
         [(UIWebView *)self.webView stringByEvaluatingJavaScriptFromString:notifyJS];
     } else {
         [self.webViewEngine evaluateJavaScript:notifyJS completionHandler:nil];
     }
 }
 
+-(void) upstream:(CDVInvokedUrlCommand *)command
+{
+    NSString* receiver = [NSString stringWithFormat:@"%@%@", FCM_PROJECT_SENDER_ID, FCM_SERVER_CONNECTION];
```
Comment (yatharthranjan) [inline src/ios/FCMPlugin.m]: Add a check if Sender ID is set or not. If not set throw an error with the appropriate message

File: src/ios/FCMPlugin.m [hunk 2]
```diff
@@ -17,6 +17,11 @@ @implementation FCMPlugin
 
 static NSString *notificationCallback = @"FCMPlugin.onNotificationReceived";
 static NSString *tokenRefreshCallback = @"FCMPlugin.onTokenRefreshReceived";
+
+static NSString *FCM_SERVER_CONNECTION = @"@gcm.googleapis.com";
+static NSString *FCM_PROJECT_SENDER_ID;
+static long long ttl = 900;
```
Comment (yatharthranjan) [inline src/ios/FCMPlugin.m]: why `long long` ?

File: src/ios/AppDelegate+FCMPlugin.m [hunk 3]
```diff
@@ -298,6 +305,16 @@ - (void)applicationDidEnterBackground:(UIApplication *)application
 }
 // [END disconnect_from_fcm]
 
+- (void)sendDataMessageFailure:(NSNotification *)notification 
+{
+    NSLog(@"Send fail");
```
Comment (yatharthranjan) [inline src/ios/AppDelegate+FCMPlugin.m]: Maybe log the content of the notification too and if any error messages for the failure ?

File: src/ios/FCMPlugin.m (hunk 2 above)
Comment (mpgxvii) [reply src/ios/FCMPlugin.m]: I think I ran into an error and had to change it to that. Let me fix it.

File: src/ios/AppDelegate+FCMPlugin.m (hunk 3 above)
Comment (mpgxvii) [reply src/ios/AppDelegate+FCMPlugin.m]: I've changed it to also log the notification object. I believe this should also include the error (in the `userInfo` property).

**Review by yatharthranjan — APPROVED** [review]

**Pull request merged**